### PR TITLE
Milestone: Add and remove tasks

### PR DESCRIPTION
**Pull request by Timbar09**
**Changes**

- Created a new JavaScript file for functions
- Implemented a function for adding a new task
- Implemented a function for deleting a task and updating all remaining tasks' indexes in the tasks array
- Implemented a function for editing task descriptions
-  Saved all changes to local storage

### CODE REVIEW

**Review by youmari — COMMENTED** [review]
Hi @Timbar09 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a future features section in the readme file 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Timbar09** [comment]
Ok.

File: src/modules/taskClass.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default class Task {
+  constructor(arr, description) {
+    this.description = description;
+    this.complete = false;
+    this.index = arr.length;
```
Comment (cynthiainga) [inline src/modules/taskClass.js]: - [ ] The property index set to the value of the new array length (i.e. if a 4th task is added to the list, the index of that task should equal to 4) which means the index should start by 1 and not 0. Please kindly fix that 🙃 😉 

Screen Shot 2023-01-20 at 18 41 42

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Timbar09, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Correct Github flow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comment under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/taskClass.js (hunk 1 above)
Comment (Timbar09) [reply src/modules/taskClass.js]: I got you @cynthiainga

Comment (Whoistolu) [reply src/modules/taskClass.js]: - Great job updating the index 👍🏽 

Screenshot 2023-01-20 at 17 28 23

**Review by Whoistolu — APPROVED** [review]
### Hi @Timbar09 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/taskClass.js (hunk 1 above)
Comment (Timbar09) [reply src/modules/taskClass.js]: Thank you 🙏 @Whoistolu

**Pull request merged**